AEPTarget/Sources/Target.swift

```diff
@@ -845,30 +869,90 @@ public class Target: NSObject, Extension {
         }
 
         guard let payloadJson = analyticsJson[TargetConstants.TargetJson.ANALYTICS_PAYLOAD] as? [String: String] else {
             return nil
         }
 
-        for (k, v) in payloadJson {
-            payload["&&\(k)"] = v
+        return payloadJson
+    }
+
+    /// Extracts click metric info from the Target response if available.
+    /// - Parameters:
+    ///     - mboxJson: Mbox dictionary.
+    /// - Returns: tuple containing `String` click token and `Dictionary` containing click tracking analytics payload.
+    private func extractClickMetric(mboxJson: [String: Any]) -> (token: String?, analyticsPayload: [String: String]?) {
+        guard let metrics = mboxJson[TargetConstants.TargetJson.METRICS] as? [[String: Any]?] else {
+            return (nil, nil)
         }
 
-        if !sessionId.isEmpty {
-            payload[TargetConstants.TargetJson.SESSION_ID] = sessionId
+        for metricItem in metrics {
+            guard
+                let metric = metricItem,
+                TargetConstants.TargetJson.MetricType.CLICK == metric[TargetConstants.TargetJson.Metric.TYPE] as? String,
+                let token = metric[TargetConstants.TargetJson.Metric.EVENT_TOKEN] as? String,
+                !token.isEmpty
+            else {
+                continue
+            }
+
+            let analyticsPayload = getAnalyticsForTargetPayload(json: metric)
+            return (token, analyticsPayload)
+        }
+
+        return (nil, nil)
+    }
+
+    /// Packages response tokens and analytics payload returned in Target response.
+    ///
+    /// If click tracking is enabled for the mbox, analytics payload returned inside metrics for click is also added.
+    /// - Parameters:
+    ///     - responseTokens: dictionary containing response tokens.
+    ///     - analyticsPayload: dictionary containing analytics for target (a4t) payload.
+    ///     - metricsAnalyticsPayload: dictionary containing a4t payload for click metric.
+    /// - Returns: `Dictionary` containing Target payload or nil.
+    private func packageMboxResponsePayload(responseTokens: [String: String]?,
+                                            analyticsPayload: [String: String]?,
+                                            metricsAnalyticsPayload: [String: String]?)
+    -> [String: Any]? {
+        var responsePayload: [String: Any] = [:]
+
+        if let responseTokens = responseTokens {
+            responsePayload[TargetConstants.TargetResponse.RESPONSE_TOKENS] = responseTokens
+        }
+
+        if let analyticsPayload = analyticsPayload {
+            responsePayload[TargetConstants.TargetResponse.ANALYTICS_PAYLOAD] = analyticsPayload
+        }
+
+        if let metricsAnalyticsPayload = metricsAnalyticsPayload {
+            responsePayload[TargetConstants.TargetResponse.CLICK_METRIC_ANALYTICS_PAYLOAD] = metricsAnalyticsPayload
         }
 
-        return payload
+        return responsePayload.isEmpty ? nil : responsePayload
```

Comment:
This is a really big class. Probably we can move all the parsing functions to separate struct. May be later.

Reply:
That's a good suggestion! We should definitely look into extracting response parsing functionality out of this class sometime to keep the code manageable here.